biogrid/libs/biogrid-simulator/src/lib/biogrid-state/biogrid-state-class.ts

```diff
@@ -1,66 +1,175 @@
-    return this.graphIndexToVertex[ind];
+  /**
+   * setnewStateGraph
+   */
+  public setnewStateGraph(newGraph: graphlib.Graph) {
+    this.graph = newGraph;
   }
 
   /**
-   * Get all GridItem positions by their index in the graph
+   * Method finds all the vertices in the graph and returns tehir names
+   * @returns the names of the vertices / gridItems in the grid
    */
-  public getAllPositionsByIndex() {
-    return this.graphIndexToVertex.map((vertex) => vertex.getRelativePosition());
+  public getAllVertices(): string[] {
+    return this.graph.nodes();
   }
 
   /**
-   * Take in a vertex and connect it to every other vertex currently present in the grid
-   * This is used in the creation of the MST
+   * Method searches the graph for a specific node by its name
+   * @param name is the string that represents the GridItem you are searching for
    */
-  private addCompletelyConnectedVertex(
-    newVertexIndex: number,
-    vertex: StateGraphVertex
-  ) {
-    const edges: StateGraphEdge[] = [];
-    for (let i = 0; i < this.graph.V; i++) {
-      if (newVertexIndex !== i) {
-        edges.push({
-          toIndex: i,
-          weight: this.calculateDistance(vertex, this.graphIndexToVertex[i]),
-          label: `${newVertexIndex}-to-${i}`,
-        });
+  public getGridItem(name: string): GridItem {
+    return this.graph.node(name);
+  }
+
+  /**
+   * getAllGridItems searches the graph vertices and retrives the gridItems which are stored on the vertices
+   */
+  public getAllGridItems(): GridItem[] {
+    return this.getAllVertices().map(vertexName => this.getGridItem(vertexName));
+  }
+
+  /**
+   * Method returns the shortest distance from every edge to the all the other edges
+   * @returns the shortest distance from any edge to the other edges
+   */
+  public getShortestDistances():{ [source: string]: { [node: string]: graphlib.Path } }  {
+    return graphlib.alg.dijkstraAll(this.graph, this.getWeightbyGraph(this.graph));
+  }
+
+  private getWeightbyGraph(graph: graphlib.Graph) {
+    return function(edge: graphlib.Edge): Distance {
+      return graph.edge(edge);
+    }
+  }
+
+  /**
+   * When given a specific edge, return the weight or distance between the two vertices
+   * @param edge is the Edge of the graph which you would like to get
+   * @returns the weight or the distance between the vertices of @param edge
+   */
+  public getWeightEdge(edge: graphlib.Edge): Distance {
+    return this.graph.edge(edge);
+  }
+
+  /**
+   * Get all GridItem positions in the graph
+   */
+  public getAllPositions(): ItemPosition[] {
+    return (this.graph.nodes() as string[]).map(vertex => this.getGridItem(vertex).getRelativePosition());
+  }
+
+  /**
+   * Add Edge when there is supposed to be an edge
+   * Add edges from the grid, to the every other part of the grid except solar panels
+   * Add reverse edges from the batteries to the grid
+   * Add edge from solar panels to the grid, not the reverse
```

Comment:
awesome comment, maybe instead of `addEdge`, the name could be `connectNewVertex` or something along those lines. Add edge may imply just adding an edge between 2 vertices

Reply:
+1